ed25519/ed25519_verify/src/ed25519_verify.rs

```diff
@@ -0,0 +1,91 @@
+extern crate ed25519_dalek;
+extern crate sha2;
+
+use ed25519_dalek::{Signature,PublicKey};
+use sha2::Sha512;
+
+
+#[derive(Debug)]
+pub struct Error(pub &'static str);
+
+impl From<&'static str> for self::Error {
+  fn from(val: &'static str) -> Self {
+    self::Error(val)
+  }
+}
+
+pub fn verify(input: &[u8; 128]) -> Result<[u8; 4], self::Error> {
+  let message = &input[..32];
+
+  let public_key = PublicKey::from_bytes(&input[32..64]).expect("public key should be correctly formed");
+  let sig = Signature::from_bytes(&input[64..128]).unwrap();//expect("signature should be correctly formed");
+
+  if public_key.verify::<Sha512>(&message, &sig).is_ok() {
```

Comment:
I think this interface should return a boolean. And the caller should translate that into 0x00000000 vs 0xffffffff.